src/interface/paint_path.c

```diff
@@ -0,0 +1,775 @@
+            }
+        }
+    }
+
+    if (RCT2_GLOBAL(0x009DEA50, sint16) != -1) {
+        sint32 staffIndex = RCT2_GLOBAL(0x009DEA50, sint16);
+        uint8 staffType = staffIndex & 0x7FFF;
+        sint16 x = RCT2_GLOBAL(0x009DE56A, sint16), y = RCT2_GLOBAL(0x009DE56E, sint16);
+
+        uint8 patrolColour = COLOUR_LIGHT_BLUE;
+
+        x = (x & 0x1F80) >> 7;
+        y = (y & 0x1F80) >> 1;
+        int offset = (x | y) >> 5;
+        int bitIndex = (x | y) & 0x1F;
+
+        bool do_it = false;
+
+        if (staffIndex >= 0) {
+            rct_peep * staff = GET_PEEP(staffIndex);
+            if (RCT2_ADDRESS(RCT2_ADDRESS_STAFF_PATROL_AREAS + (staff->staff_id * 512), uint32)[offset] & (1 << bitIndex)) {
+                do_it = true;
+            } else {
+                patrolColour = COLOUR_GREY;
+                staffType = (staff->staff_type + 200) * 512;
+                if (RCT2_ADDRESS(RCT2_ADDRESS_STAFF_PATROL_AREAS + staffType, uint32)[offset] & (1 << bitIndex)) {
+                    do_it = true;
+                }
+            }
+        } else {
+            staffType = (staffType + 200) * 512;
+            if (RCT2_ADDRESS(RCT2_ADDRESS_STAFF_PATROL_AREAS + staffType, uint32)[offset] & (1 << bitIndex)) {
+                do_it = true;
+            }
+        }
+
+        if (do_it) {
+            uint32 imageId = 2618;
+            int height = map_element->base_height * 8;
+            if (map_element->properties.path.type & 0x04) {
+                imageId = 2619 + ((map_element->properties.path.type + get_current_rotation()) & 3);
+                height += 16;
+            }
+
+            sub_98196C(imageId | patrolColour << 19 | 0x20000000, 16, 16, 1, 1, 0, height + 2, get_current_rotation());
+        }
+    }
+
+
+    if (gCurrentViewportFlags & VIEWPORT_FLAG_PATH_HEIGHTS) {
+        uint16 height = 3 + map_element->base_height * 8;
+        if (map_element->properties.path.type & 0x04) {
+            height += 8;
+        }
+        uint32 imageId = (SPR_HEIGHT_MARKER_BASE + height / 16) | COLOUR_GREY << 19 | 0x20000000;
+        imageId += RCT2_GLOBAL(RCT2_ADDRESS_CONFIG_HEIGHT_MARKERS, uint16);
+        imageId -= RCT2_GLOBAL(0x01359208, uint16);
+        sub_98196C(imageId, 16, 16, 1, 1, 0, height, get_current_rotation());
+    }
+
+    uint8 pathType = (map_element->properties.path.type & 0xF0) >> 4;
+    rct_path_type * dword_F3EF6C = (rct_path_type *) object_entry_groups[OBJECT_TYPE_PATHS].chunks[pathType];
+
+    if (dword_F3EF6C->var_0A == 0) {
+        loc_6A37C9(map_element, height, dword_F3EF6C, word_F3F038, dword_F3EF70, dword_F3EF74);
+    } else {
+        loc_6A3B57();
+        RCT2_GLOBAL(0x00F3EF6C, uint32) = (uint32) dword_F3EF6C;
+        RCT2_GLOBAL(0x00F3F038, uint16) = word_F3F038;
+        RCT2_GLOBAL(0x00F3EF70, uint16) = dword_F3EF70;
+        RCT2_GLOBAL(0x00F3EF74, uint16) = dword_F3EF74;
+        RCT2_CALLPROC_X(0x6A3B57, 0, 0, get_current_rotation(), height, (int) map_element, 0, 0);
```

Comment:
Can you add a little not stating that this is just for running the old code. I got a little confused wondering why there was a double function call here.